### ISSUE TITLE
Allow lazy loading of sha256

### ISSUE THREAD

**Issue by tresf** [qzind/tray#186]
Currently we require a strict import order, `sha-256` must come BEFORE `qz-tray` since we [configure `hash` property to `Sha256` in the constructor](https://github.com/qzind/tray/blob/2.0/js/qz-tray.js#L459).

Since initialization doesn't require sha256, we should be safe checking for it slightly later, such as through a `getter`.

This would allow the import of `qz-tray.js` to come before `sha-256.min.js`.

```html
	<script type="text/javascript" src="sha-256.min.js"></script>
	<script type="text/javascript" src="qz-tray.js"></script>
```

**Comment by bberenz** [comment]
Implemented in e4ce6008aa40632605c0af31f2d774651c5fd9ef
Wrapped in a function, so it doesn't have to exist at load for assignment (like we do for promises)

**Comment by tresf** [comment]
Closing as fixed.  We'll report any issues with 2.0.4 as they arise.
